### PR TITLE
Implemented task categories

### PR DESCRIPTION
**Pull request by sosaagbontaen**
When users create a new task, they can now create and select a category that it should belong to. Similarly, users can also change a task's category when editing the task information.
Kapture 2022-07-21 at 10 21 50

### CODE REVIEW

File: CentralityiOS/CentralityiOS/ViewControllers/CategoryModalViewController.m [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        }];
+        
+        completionHandler(YES);
+        
+    }];
+    
+    deleteAction.backgroundColor = [UIColor systemRedColor];
+    
+    UISwipeActionsConfiguration *swipeActions = [UISwipeActionsConfiguration configurationWithActions:@[deleteAction]];
+    swipeActions.performsFirstActionWithFullSwipe=false;
```
Comment (mayguan) [inline CentralityiOS/CentralityiOS/ViewControllers/CategoryModalViewController.m]: -> NO

File: CentralityiOS/CentralityiOS/ViewControllers/CategoryModalViewController.m [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    return swipeActions;
+}
+
+- (IBAction)addCategoryAction:(id)sender {
+    if ([self.nameOfCategoryToAdd.text isEqual: @""]){
```
Comment (mayguan) [inline CentralityiOS/CentralityiOS/ViewControllers/CategoryModalViewController.m]: -> isEqualToString

File: CentralityiOS/CentralityiOS/ViewControllers/CategoryModalViewController.m [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    }
+    
+    CategoryObject *newCategory = [CategoryObject new];
+    newCategory.categoryName = self.nameOfCategoryToAdd.text;
+    newCategory.owner = [PFUser currentUser];
+    
+    [newCategory saveInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
+        if (succeeded) {
+            [self fetchCategories];
+            self.nameOfCategoryToAdd.text = @"";
```
Comment (mayguan) [inline CentralityiOS/CentralityiOS/ViewControllers/CategoryModalViewController.m:113]: Why do we need to set it to an empty string?

File: CentralityiOS/CentralityiOS/ViewControllers/EditTaskModalViewController.m [hunk 4]
```diff
@@ -12,13 +12,31 @@
...
 
+
 @implementation EditTaskModalViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
     self.taskNameInput.text = self.taskFromFeed.taskTitle;
     self.taskDescInput.text = self.taskFromFeed.taskDesc;
+    if (self.taskCategory){
+    [self.changeCategoryButton setTitle:self.taskCategory.categoryName forState:UIControlStateNormal];
```
Comment (mayguan) [inline CentralityiOS/CentralityiOS/ViewControllers/EditTaskModalViewController.m]: nit: indentation

**Review by mayguan — APPROVED** [review]

File: CentralityiOS/CentralityiOS/ViewControllers/CategoryModalViewController.m (hunk 3 above)
Comment (sosaagbontaen) [reply CentralityiOS/CentralityiOS/ViewControllers/CategoryModalViewController.m:113]: This is just preference. I figured it makes more sense to clear the field after the user adds a new category so they don't accidentally add the same category twice. Here's a demo of the feature : 
Kapture 2022-07-21 at 15 35 54

**Pull request merged**